src/main/java/io/pravega/connectors/hadoop/PravegaInputFormat.java

```diff
@@ -66,18 +77,20 @@
         final String deserializerClassName = Optional.ofNullable(conf.get(PravegaInputFormat.DESERIALIZER)).orElseThrow(() ->
                 new IOException("The event deserializer must be configured (" + PravegaInputFormat.DESERIALIZER + ")"));
 
+        if (this.clientFactory == null) {
+            this.clientFactory = ClientFactory.withScope(scopeName, controllerURI);
+        }
         // generate a split per segment
         List<InputSplit> splits = new ArrayList<InputSplit>();
-        try (ClientFactory clientFactory = ClientFactory.withScope(scopeName, controllerURI)) {
-            BatchClient batchClient = clientFactory.createBatchClient();
-
-            for (Iterator<SegmentInfo> iter = batchClient.listSegments(new StreamImpl(scopeName, streamName)); iter.hasNext(); ) {
-                SegmentInfo segInfo = iter.next();
-                Segment segment = segInfo.getSegment();
-                PravegaInputSplit split = new PravegaInputSplit(segment, segInfo.getStartingOffset(), segInfo.getWriteOffset());
-                splits.add(split);
-            }
+        BatchClient batchClient = clientFactory.createBatchClient();
+        for (Iterator<SegmentInfo> iter = batchClient.listSegments(new StreamImpl(scopeName, streamName)); iter.hasNext(); ) {
+            SegmentInfo segInfo = iter.next();
+            Segment segment = segInfo.getSegment();
+            PravegaInputSplit split = new PravegaInputSplit(segment, segInfo.getStartingOffset(), segInfo.getWriteOffset());
+            splits.add(split);
         }
+        this.clientFactory.close();
+        this.clientFactory = null;
```

Comment:
In the test scenario where a client factory is supplied, you should probably not 'throw it away' here.

Reply:
oooooh, yes, you are correct. accepted